cas/remote.go

```diff
@@ -125,14 +125,41 @@ func (r Remote) Store(ds Store, aci io.Reader) (*Remote, error) {
 	return &r, nil
 }
 
+// downloadACI gets the aci specified at aciurl
 func downloadACI(ds Store, aciurl string) (*os.File, error) {
-	res, err := http.Get(aciurl)
+	return downloadHTTP(aciurl, "ACI", ds.tmpFile)
+}
+
+// downloadSignatureFile gets the signature specified at sigurl
+func downloadSignatureFile(sigurl string) (*os.File, error) {
+	getTemp := func() (*os.File, error) {
+		return ioutil.TempFile("", "")
+	}
+
+	return downloadHTTP(sigurl, "signature", getTemp)
+}
+
+// downloadHTTP retrieves url, creating a temp file using getTempFile
+// file:// http:// and https:// urls supported
+func downloadHTTP(url, label string, getTempFile func() (*os.File, error)) (*os.File, error) {
+	tmp, err := getTempFile()
```

Comment:
Why pass in a factory function if you just immediately call it? Wouldn't it be easier to just pass in *os.File?